chain/network/src/routing.rs

```diff
@@ -367,6 +387,79 @@ impl RoutingTable {
         })
     }
 
+    /// If peer_id is not on memory check if it is on disk in bring it back on memory.
+    fn touch(&mut self, peer_id: &PeerId) {
+        if peer_id == self.peer_id() || self.peer_last_time_reachable.contains_key(peer_id) {
+            return;
+        }
+
+        let me = self.peer_id().clone();
+
+        if let Ok(Some(nonce)) =
+            self.store.get_ser::<u64>(COL_PEER_COMPONENT, Vec::from(peer_id.clone()).as_ref())
+        {
+            let mut update = self.store.store_update();
+            let enc_nonce = u64_as_bytes(nonce);
+
+            if let Ok(Some(edges)) =
+                self.store.get_ser::<Vec<Edge>>(COL_COMPONENT_EDGES, enc_nonce.as_ref())
+            {
+                for edge in edges.into_iter() {
+                    for &peer_id in vec![&edge.peer0, &edge.peer1].iter() {
+                        if peer_id == &me || self.peer_last_time_reachable.contains_key(peer_id) {
+                            continue;
+                        }
+
+                        if let Ok(Some(cur_nonce)) = self
+                            .store
+                            .get_ser::<u64>(COL_PEER_COMPONENT, Vec::from(peer_id.clone()).as_ref())
```

Comment:
It's better to extract the part that reads from the store into a helper function to make the code more readable.

Reply:
done